### PR TITLE
Add a script to measure build performance

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

In order to avoid redundancy and make it more maintainable, I refactored the performance script.
  - I moved the logic into a common script file (kani-perf-setup.sh)
  - I also changed the performance to run in a temporary file
    - To simplify deletion
  - I changed benchcomp to work with the new location

### Resolved issues:

Related to #2442 

### Related RFC:

Optional #ISSUE-NUMBER.

### Call-outs:

-  I haven't actually created the CI job or the benchcomp configuration / parser.
- @karkhaz does a benchcomp parser have access to the script output? I guess I could write things the output into a file... Ideally, I would like to get the path from the script output.

### Testing:

* How is this change tested? Manually ran the regression and the parser.

* Is this a refactor change? Kinda

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

**Comment by celinval** [comment]
@karkhaz can you help me out please? It seems that I cannot rename the output folder because it will stop working for previous versions of Kani. Which `kani-perf.sh` is benchcomp running?? Does this mean that I won't be able to add a new benchmark job that runs on older versions of Kani since the script won't exist there?

File: scripts/kani-perf-setup.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+run_benchmarks() {
+  suite=$(basename "${SUITE}")
+  mode="cargo-kani-test"
+  echo "Check compiletest suite=$suite mode=$mode"
+  cargo run -p compiletest -- --suite $suite --mode $mode --no-fail-fast --report-time "$@"
+}
+
+print_result() {
+  exit_code=$1
```
Comment (karkhaz) [inline scripts/kani-perf-setup.sh:45]: Is this intentional?

**Review by karkhaz — APPROVED** [review]
(Had a discussion offline about benchcomp).

Thanks very much for this, looking forward to having less duplication between CI and local invocations.

File: scripts/kani-perf-setup.sh (hunk 1 above)
Comment (celinval) [reply scripts/kani-perf-setup.sh:45]: Yes. A bunch of these variables should be local, but I don't think it really matters here.

**Comment by celinval** [comment]
Tbh, this solution became too hacky due to benchcomp configuration. I'll create a simple PR without any change to perf.

**Pull request closed without merging**